### ISSUE TITLE
[QAIT] Incorrect dependencies Error at time of running command: php magento setup:di:compile

### ISSUE THREAD

**Issue by ramanjeetaulakh-qait** [oxipay/oxipay-magento-2.x#28]
Steps:
1) Installed magento 2.x
2) Created code folder under var/www/html/mage_2/app
3) Copied oxipay plug in zip under var/www/html/mage_2/app/code and Unzipped
4) Deleted zipped file
5) Run command: php magento setup:upgrade under /var/bin folder
6) Run command: php magento setup:di:compile under /var/www/html/mage_2/bin
after step 6 

Result:
Getting error as Incorrect dependencies. 

Expected:
There should be no error.

Screen cast: http://screencast-o-matic.com/watch/cbhf23X3OS

Tech Stack:
Ubuntu (16.04 LTS)
PHP version:(5.6.30)
mysql: (14.14)
Apache: (2.4.18)
Magento (ver 2.1.5)
Oxipay_magento2.x (v0.0.14)

Error:
Compilation was started.
Interception cache generation... 6/7 [========================>---]  85% 54 secs 454.2 MiBErrors during compilation:
        Oxipay\OxipayPaymentGateway\Controller\Checkout\AbstractAction
                Incorrect dependency in class Oxipay\OxipayPaymentGateway\Controller\Checkout\AbstractAction in /var/www/html/mage_2/app/code/Oxipay/OxipayPaymentGateway/Controller/Checkout/AbstractAction.php
\Magento\Framework\Message\ManagerInterface already exists in context object
        Oxipay\OxipayPaymentGateway\Model\Ui\ConfigProvider
                Incorrect dependency in class Oxipay\OxipayPaymentGateway\Model\Ui\ConfigProvider in /var/www/html/mage_2/app/code/Oxipay/OxipayPaymentGateway/Model/Ui/ConfigProvider.php
\Magento\Framework\App\Config\ScopeConfigInterface already exists in context object
Total Errors Count: 2
Errors during compilation:
        Oxipay\OxipayPaymentGateway\Controller\Checkout\AbstractAction
                Incorrect dependency in class Oxipay\OxipayPaymentGateway\Controller\Checkout\AbstractAction in /var/www/html/mage_2/app/code/Oxipay/OxipayPaymentGateway/Controller/Checkout/AbstractAction.php
\Magento\Framework\Message\ManagerInterface already exists in context object
        Oxipay\OxipayPaymentGateway\Model\Ui\ConfigProvider
                Incorrect dependency in class Oxipay\OxipayPaymentGateway\Model\Ui\ConfigProvider in /var/www/html/mage_2/app/code/Oxipay/OxipayPaymentGateway/Model/Ui/ConfigProvider.php
\Magento\Framework\App\Config\ScopeConfigInterface already exists in context object
Total Errors Count: 2
Errors during compilation:
        Oxipay\OxipayPaymentGateway\Controller\Checkout\AbstractAction
                Incorrect dependency in class Oxipay\OxipayPaymentGateway\Controller\Checkout\AbstractAction in /var/www/html/mage_2/app/code/Oxipay/OxipayPaymentGateway/Controller/Checkout/AbstractAction.php
\Magento\Framework\Message\ManagerInterface already exists in context object
        Oxipay\OxipayPaymentGateway\Model\Ui\ConfigProvider
                Incorrect dependency in class Oxipay\OxipayPaymentGateway\Model\Ui\ConfigProvider in /var/www/html/mage_2/app/code/Oxipay/OxipayPaymentGateway/Model/Ui/ConfigProvider.php
\Magento\Framework\App\Config\ScopeConfigInterface already exists in context object
Total Errors Count: 2
Interception cache generation... 7/7 [============================] 100% 2 mins 401.8 MiB
Generated code and dependency injection configuration successfully.

**Comment by esteban-terem** [comment]
Hi. Although running `setup:di:compile` is not part of the installation procedure, this should not fail.
A PR (#32) was created to merge commit 23502e9, which resolves this issue (branch issue-28). Thanks.

**Comment by blueninja2** [comment]
As per comment from Esteban, we'll be moving this to the To-Do list (appears to be resolved as above).

**Comment by ramanjeetaulakh-qait** [comment]
QA Team verified this issue with latest oxipay-Plugin build(v0.0.16) and found as Fixed.
Screen Cast: http://screencast-o-matic.com/watch/cbhlX6XUMe
